sonar-plugin-api/src/main/java/org/sonar/api/profiles/ProfileDefinition.java

```diff
@@ -24,9 +24,33 @@
 import org.sonar.api.utils.ValidationMessages;
 
 /**
- * Define a profile which is automatically registered during sonar startup.
- * The components <code>AnnotationProfileParser</code> and <code>XMLProfileParser</code> can be used to help implementing the method create().
+ * Define a quality profile which is automatically registered during SonarQube startup (ie built-in quality profile).
+ * The component {@link XMLProfileParser} can be used to help implementing the method create().
+ * <p>
+ * Example:
+ * <pre>
+ * public class MyProfile extends ProfileDefinition {
+ * 
+ *   private final RuleFinder ruleFinder;
+ * 
+ *   public MyProfile(RuleFinder ruleFinder) {
+ *     this.ruleFinder = ruleFinder;
+ *   }
+ * 
+ *   {@literal @}Override
+ *   public RulesProfile createProfile(ValidationMessages validation) {
+ *     final RulesProfile profile = RulesProfile.create("My Profile", "java");
+
+ *     // activate rules providing the severity (can be different that rule default severity)
+ *     profile.activateRule(Rule.create("checkstyle", "UnusedVariable"), Severity.MAJOR);
+ *     // keep default rule severity
+ *     profile.activateRule(ruleFinder.findByKey("checkstyle", "UnusedParameter"), null);
```

Comment:
Just a detail : using RuleFinder.findByKey() will generate on SQL query each time it's called. Would be best to advice the usage of RuleFinder.findAll(RuleQuery).
I know it's a technical detail, but ProfileDefinition API should be replaced by another API without having to do query db to define a profile.
Waiting for that, if we could prevent having to much SQL queries generated when registrering a profile, it would be cool.